crates/key-server/src/server.rs

```diff
@@ -255,7 +257,18 @@ impl Server {
             .await
             .map_err(|e| {
                 warn!("Dry run execution failed ({:?}) (req_id: {:?})", e, req_id);
-                InternalError::Failure
+                // A dry run will fail if called with a newly created object parameter that the FN has not yet seen.
+                // In that case, the user gets a 503 status code (service unavailable).
+                match e {
+                    Error::RpcError(ClientError::Call(e)) => match e.code() {
+                            INVALID_PARAMS_CODE => {
```

Comment:
are there other interesting errors we should handle?

Reply:
At this level, these are the possible errors: https://github.com/paritytech/jsonrpsee/blob/6540970ddfa5e0c84cc0ec8567a32233dfa0591a/types/src/error.rs#L121-L147. I don't see anything we specifically need to consider, but we could monitor the error metrics to see, if we've missed something.

Reply:
maybe METHOD_NOT_FOUND_CODE?

Reply:
What would be the error there? That the seal_approve functions doesn't exist?

Reply:
yes

Reply:
Added, returns an InvalidPTB error in that case.